addons/mail/static/src/chatter/web_portal/composer_patch.js

```diff
@@ -5,7 +5,9 @@ import { patch } from "@web/core/utils/patch";
 patch(Composer.prototype, {
     get placeholder() {
         if (this.thread && this.thread.model !== "discuss.channel" && !this.props.placeholder) {
-            if (this.props.type === "message") {
+            if (this.thread.isReadonly()) {
+                return _t("You don't have the rights to post on this Document...");
+            } else if (this.props.type === "message") {
```

Comment:
As I said in previous comment: should deal with `isReadonly`/`canPost` with value `undefined`, which is when there's need to fetch and we don't know the value. This allow to show a spinner while it's being fetched, so that we don't lie on whether we can post or not.

I would also word to `You do not have the rights to post on this Document.`:
- "do not" instead of "don't", to put more emphasis we do not have access rights and it's a formality (no rights)
- No "..." as this is a clear instruction. `...` in other placeholder are there to let the user write in composer. There's no `...` if the user cannot post a message.